### PR TITLE
Adding prefix to shop order number when it has duplicate

### PR DESCRIPTION
**Pull request by jnmrclmbsjse**
This PR includes:
1. Adding prefix to shop order number when duplicate order number is found e.g `12(1)`.
2. Added maximum tries for prefixing.
3. Unit test.

### CODE REVIEW

**Comment by jnmrclmbsjse** [comment]
Task linked: [CU-8692x16zw Order duplicate number](https://app.clickup.com/t/8692x16zw)

File: src/StoreKeeper/WooCommerce/B2C/Exports/OrderExport.php [hunk 1]
```diff
@@ -733,19 +755,6 @@ private function getStorekeeperIdBySku($sku): ?int
         return null;
     }
 
-    protected function convertKnownGeneralException(GeneralException $throwable): Throwable
-    {
-        if ('ShopModule::OrderDuplicateNumber' === $throwable->getApiExceptionClass()) {
-            return new ExportException(
-                esc_html__('Order with this order number already exists.', I18N::DOMAIN),
```
Comment (ikulis) [inline src/StoreKeeper/WooCommerce/B2C/Exports/OrderExport.php:740]:
```
esc_html__('Order with this order number already exists. Tried duplicate up to {$shopOrderId}(self::MAXIMUM_DUPLICATE_COUNT)', I18N::DOMAIN),
```

I think it's still good to keep the error

File: tests/unit/Exports/OrderExportTest.php [hunk 2]
```diff
@@ -879,10 +880,8 @@ function () {
         try {
             $this->processTask($task);
             $this->assertEquals("$shopOrderId{$expectedPrefix}", $sent_order['shop_order_number'], 'Shop order number sent should have expected prefix');
-        } catch (GeneralException $exception) {
-            if ('ShopModule::OrderDuplicateNumber' === $exception->getApiExceptionClass()) {
-                $failed = true;
-            }
+        } catch (ExportException $exception) {
+            $failed = true;
```
Comment (ikulis) [inline tests/unit/Exports/OrderExportTest.php:884]:
```suggestion
            $failed = 'ShopModule::OrderDuplicateNumber' === $exception->getPrevious()->getApiExceptionClass();
```

Comment (jnmrclmbsjse) [reply tests/unit/Exports/OrderExportTest.php]: Changes implemented.

**Pull request merged**